contractsdk/cpp/xchain/syscall.cc

```diff
@@ -14,9 +15,10 @@ static bool syscall_raw(const std::string& method, const std::string& request,
     if (response_len <= 0) {
         return true;
     }
+    std::vector<char> buf(response_len);
```

Comment:
Why do you need a vector as a buffer here?

Reply:
The expression `response->data()` returns a `const char*`,  changing a const char* is an undefined behaviors.  std::vector<char> or char *buf = new (std::nothrow) char[response_len + 1] all work here.

Reply:
Maybe `&(*response)[0]` the address of first element in `response` will be a better choice.
https://stackoverflow.com/a/16502000